### PR TITLE
[stdlib] Fix `String.splitlines` and test and move to string_slice

### PR DESCRIPTION
**Pull request by martinvuyk**
Fix `String.splitlines` and test and move to string_slice

### CODE REVIEW

File: stdlib/src/utils/string_slice.mojo [hunk 1]
```diff
@@ -192,6 +192,44 @@ fn _is_valid_utf8(data: UnsafePointer[UInt8], length: Int) -> Bool:
...
 
 
+fn _isnewline_start(
+    ptr: UnsafePointer[UInt8], read_ahead: Int = 1
+) -> (Bool, Int):
+    """Returns if the first item in the pointer is the start of
+    a newline sequence, and its length.
+    """
+    # TODO add line and paragraph separator as stringliteral
+    # once unicode escape secuences are accepted
```
Comment (JoeLoser) [inline stdlib/src/utils/string_slice.mojo:202]:
```suggestion
    # once Unicode escape sequences are accepted
```

File: stdlib/src/utils/string_slice.mojo [hunk 2]
```diff
@@ -192,6 +192,44 @@ fn _is_valid_utf8(data: UnsafePointer[UInt8], length: Int) -> Bool:
     return _validate_utf8_simd_slice[4, True](ptr, length, iter_len) == 0
 
 
+fn _isnewline_start(
+    ptr: UnsafePointer[UInt8], read_ahead: Int = 1
+) -> (Bool, Int):
+    """Returns if the first item in the pointer is the start of
+    a newline sequence, and its length.
+    """
+    # TODO add line and paragraph separator as stringliteral
```
Comment (JoeLoser) [inline stdlib/src/utils/string_slice.mojo:201]:
```suggestion
    # TODO add line and paragraph separator as StringLiteral
```

**Review by JoeLoser — APPROVED** [review]
Looks good.  Once the few minor comments are addressed, happy to sync internally and land this.  Thank you!

File: stdlib/src/utils/string_slice.mojo (hunk 1 above)
Comment (martinvuyk) [reply stdlib/src/utils/string_slice.mojo:202]: damn I've already written that in other PRs :( I'll find all and fix them

**Comment by JoeLoser** [comment]
!sync

**Pull request closed without merging**